src/client/components/app/Transactions/components/TxError.tsx

```diff
@@ -1,62 +1,118 @@
+  color: inherit;
+  max-width: 100%;
+  font-size: 1.2rem;
 `;
 
 const StyledTxError = styled.div<{ errorType?: ErrorType }>`
   display: flex;
-  align-items: center;
-  justify-content: center;
-  gap: 5rem;
-  padding: 0 2.1rem;
-  font-weight: 500;
+  align-items: stretch;
+  font-weight: 100;
   font-size: 1.4rem;
   border-radius: ${({ theme }) => theme.globalRadius};
-  text-transform: uppercase;
-  min-height: 7.8rem;
-  max-height: 10rem;
-  overflow: hidden;
-  overflow-y: auto;
-  flex: 1;
+  padding: 0.5rem 0.7rem;
+  flex-direction: column;
 
-  background-color: ${({ theme }) => theme.colors.txModalColors.error};
-  color: ${({ theme }) => theme.colors.txModalColors.textContrast};
+  background-color: ${({ theme }) => theme.colors.txModalColors.error.background};
+  color: ${({ theme }) => theme.colors.txModalColors.error.color};
 
   ${({ errorType, theme }) =>
     errorType === 'warning' &&
     `
-    background-color: ${theme.colors.txModalColors.warning};
+    background-color: ${theme.colors.txModalColors.warning.background};
+    color: ${theme.colors.txModalColors.warning.color};
   `}
 `;
 
+const StyledTxErrorTitle = styled.div`
+  display: flex;
+  flex: 1;
+  gap: 0.7rem;
+  align-items: center;
+`;
+
+const StyledTxErrorDescription = styled.div<{ isExpanded: boolean }>`
+  max-height: ${({ isExpanded }) => (isExpanded ? '7rem' : '0')};
+  overflow-y: scroll;
+  margin-top: ${({ isExpanded }) => (isExpanded ? '1rem' : '0')};
+  transition: margin-top 0.5s ease-in-out, max-height 0.5s ease-in-out;
+  ::-webkit-scrollbar {
+    width: 0.2rem;
+  }
+
+  /* Track */
+  ::-webkit-scrollbar-track {
+    background: ${({ theme }) => theme.colors.txModalColors.warning.color}80;
+  }
+
+  /* Handle */
+  ::-webkit-scrollbar-thumb {
+    background: ${({ theme }) => theme.colors.txModalColors.warning.color};
+  }
+`;
+
 type ErrorType = 'error' | 'warning';
 export interface TxErrorProps {
-  errorText?: string;
+  errorDescription?: string;
+  errorTitle?: string;
   errorType?: ErrorType;
 }
 
-export const TxError: FC<TxErrorProps> = ({ errorText, errorType, children, ...props }) => {
+export const TxError: FC<TxErrorProps> = ({ errorTitle, errorDescription, errorType, children, ...props }) => {
   const { t } = useAppTranslation('common');
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  useEffect(() => {
+    //clean state once error message changes
+    setIsExpanded(false);
+  }, [errorTitle, errorDescription]);
 
   return (
     <StyledTxError errorType={errorType} {...props}>
-      <StyledIcon Component={WarningIcon} />
+      <StyledTxErrorTitle>
+        <StyledIcon Component={WarningIcon} errorType={errorType} />
+
+        <StyledTextTitle>{errorTitle || t('errors.unknown')}</StyledTextTitle>
 
-      <StyledText>{errorText || t('errors.unknown')}</StyledText>
+        {errorDescription && (
+          <StyledIcon
+            onClick={() => setIsExpanded(!isExpanded)}
```

Comment:
This might lead to unexpected results because this is asynchronous, we should use the previous state rather setting this directly

```suggestion
            onClick={() => setIsExpanded((prevIsExpanded) => !prevIsExpanded)}
```

Reply:
fixed on 43649ef98fa2d6e9d704f21f6f7892a3ba13f7f2